### PR TITLE
TCL: Remove TCL dependency from TSControl and EventControl

### PR DESCRIPTION
**Pull request by chitianhao**

### CODE REVIEW

**Comment by chitianhao** [comment]
Looks like there is an issue. I'll look it up.

File: mgmt/api/EventControlMain.cc [hunk 1]
```diff
@@ -334,7 +323,6 @@ event_callback_main(void *arg)
             if (ret == TS_ERR_NET_READ || ret == TS_ERR_NET_EOF) { // preprocess_msg FAILED!
               Debug("event", "[event_callback_main] preprocess_msg FAILED; skip!");
               remove_event_client(client_entry, accepted_clients);
```
[the remark this answers is not in the document]
Comment (chitianhao) [reply mgmt/api/EventControlMain.cc]: 👍

File: mgmt/api/EventControlMain.cc [hunk 2]
```diff
@@ -245,14 +247,8 @@ event_callback_main(void *arg)
 
   Debug("event", "[event_callback_main] listen on socket = %d", con_socket_fd);
 
-  // initialize queue for accepted con
```
[the remark this answers is not in the document]
Comment (chitianhao) [reply mgmt/api/EventControlMain.cc]: The TCL hash table requires an initialization but I don't think unordered_map needs to be created.

File: mgmt/api/EventControlMain.cc [hunk 3]
```diff
@@ -312,17 +300,17 @@ event_callback_main(void *arg)
...
       } // end if (new_con_fd >= 0 && FD_ISSET(new_con_fd, &selectFDs))
 
       // some other file descriptor; for each one, service request
       if (fds_ready > 0) { // RECEIVED A REQUEST from remote API client
         // see if there are more fd to set - iterate through all entries in hash table
-        con_entry = ink_hash_table_iterator_first(accepted_clients, &con_state);
-        while (con_entry) {
-          client_entry = (EventClientT *)ink_hash_table_entry_value(accepted_clients, con_entry);
+        for (auto it = accepted_clients.begin(); it != accepted_clients.end();) {
+          client_entry = it->second;
```
Comment (SolidWallOfCode) [inline mgmt/api/EventControlMain.cc]: For compactness you could do `client_entry = it++->second;`.

File: mgmt/api/EventControlMain.cc [hunk 4]
```diff
@@ -312,17 +300,17 @@ event_callback_main(void *arg)
...
+          it++; // prevent the breaking of remove_event_client
```
Comment (SolidWallOfCode) [inline mgmt/api/EventControlMain.cc]: Pre-increment unless you need post-increment.

**Comment by bryancall** [comment]
Cherry picked to 8.1.0

**Pull request merged**